frontend/src/components/DefenceBox/DefenceConfigurationInput.tsx

```diff
@@ -2,28 +2,51 @@ import { KeyboardEvent, useEffect, useState } from 'react';
 
 import ThemedNumberInput from '@src/components/ThemedInput/ThemedNumberInput';
 import ThemedTextArea from '@src/components/ThemedInput/ThemedTextArea';
+import { Defence, DefenceConfigItem } from '@src/models/defence';
+import { validateDefence } from '@src/service/defenceService';
 
 function DefenceConfigurationInput({
 	currentValue,
 	disabled,
 	inputType,
 	setConfigurationValue,
 	id,
+	defence,
+	config,
 }: {
 	currentValue: string;
 	disabled: boolean;
 	inputType: 'text' | 'number';
-	setConfigurationValue: (value: string) => void;
+	setConfigurationValue: (value: string) => Promise<void>;
 	id: string;
+	defence: Defence;
+	config: DefenceConfigItem;
```

Comment:
Not convinced you should be passing these two in, it gives DefenceConfigurationInput too much knowledge it shouldn't need.

You only need them so you can call validateDefence, but instead you could pass in a validate function prop that only takes the value to check, and then you would invoke validateDefence from the parent component where you already have access to defence and config.

Reply:
I tried this, but then the onChange validation doesn't seem to work. 
Might be missing something, but let's look at it together tomorrow